### PR TITLE
Fixed slider thumbWith, #3359

### PR DESCRIPTION
**Pull request by JeanPhilippeKernel**

### CODE REVIEW

**Comment by deltakosh** [comment]
Thanks a lot:) As discussed offline, please add options to keep current behavior and add this one as optional

**Comment by JeanPhilippeKernel** [comment]
okay:) no problem !

**Comment by JeanPhilippeKernel** [comment]
property added for switching between default or clamped behavior:)

**Comment by deltakosh** [comment]
Seems like you have errors on the Travis build:)

**Comment by JeanPhilippeKernel** [comment]
:( yep I don't know why

**Comment by deltakosh** [comment]
Seems like a Travis issue
I'll check what's wrong asap:)

**Comment by JeanPhilippeKernel** [comment]
okay great !! I'm restarting the build validation in my side to see too:)

**Comment by JeanPhilippeKernel** [comment]
:( local build validation for Normals failed... scriptToRun's value : "404 not found" , validation.js :240:72

**Comment by deltakosh** [comment]
OK just merge with Master and it should be good

**Comment by deltakosh** [comment]
Good job!
Now I would like to refactor code between two functions as there is a lot of duplicated code that will be hard to maintain

Keep the good work ;)

**Comment by JeanPhilippeKernel** [comment]
PR in progress for that :)

**Comment by JeanPhilippeKernel** [comment]
looks simple and maybe clear, but personally I think there is a lot of if-else checking

**Comment by deltakosh** [comment]
Looks clearly better! Let me do some tests now :)

**Comment by JeanPhilippeKernel** [comment]
okay :)

**Comment by deltakosh** [comment]
This looks good!
Let me merge it :D

File: gui/src/controls/slider.ts [hunk 1]
```diff
@@ -197,10 +224,9 @@ module BABYLON.GUI {
                     context.shadowOffsetY = this.shadowOffsetY;
                 }
 
-                // Thumb
                 if (this._isThumbCircle) {
                     context.beginPath();
-                    context.arc(left + thumbPosition, this._currentMeasure.top + this._currentMeasure.height / 2, effectiveThumbWidth / 2, 0, 2 * Math.PI);
+                    context.arc(left + thumbPosition + (effectiveThumbWidth / 2), this._currentMeasure.top + this._currentMeasure.height / 2, effectiveThumbWidth / 2, 0, 2 * Math.PI);
```
Comment (deltakosh) [inline gui/src/controls/slider.ts]: what if the thumb is clamped?
Seems like the circle must be different right?

**Comment by deltakosh** [comment]
Now you need to update the doc:
https://github.com/BabylonJS/Documentation/blob/master/content/How_To/GUI/Gui.md

(and perhaps add a small demo in the PG in the doc as well)

Thanks a lot!

**Comment by JeanPhilippeKernel** [comment]
oh :) great !! thanks so much .... okay not problem:)

**Pull request merged**